packages/compass-import-export/src/components/import-preview.tsx

```diff
@@ -109,29 +173,108 @@ function ImportPreview({
                     onFieldCheckedChanged(field.path, !!e.target.checked)
                   }
                 />
-                <div>
-                  <Label
-                    id={`toggle-import-field-label-${field.path}`}
-                    className={fieldPathHeaderStyles}
-                    htmlFor={`toggle-import-field-checkbox-${field.path}`}
-                  >
-                    <span title={field.path}>{field.path}</span>
-                  </Label>
-                  <SelectFieldType
-                    fieldPath={field.path}
-                    selectedType={field.type}
-                    onChange={(newType: string) =>
-                      setFieldType(field.path, newType)
-                    }
+                <Label
+                  id={`toggle-import-field-label-${field.path}`}
+                  className={fieldPathHeaderStyles}
+                  htmlFor={`toggle-import-field-checkbox-${field.path}`}
+                >
+                  <span title={field.path}>{field.path}</span>
+                </Label>
+              </div>
+              <div className={fieldTypeContainerStyles}>
+                {analyzed ? (
+                  <>
+                    <SelectFieldType
+                      fieldPath={field.path}
+                      selectedType={field.type}
+                      onChange={(newType: string) =>
+                        setFieldType(field.path, newType)
+                      }
+                    />
+                    {field.result && (
+                      <MixedWarning
+                        result={field.result}
+                        selectedType={field.type}
+                      />
+                    )}
+                  </>
+                ) : (
+                  <Placeholder
+                    width={spacing[3] * 9}
+                    data-testid={`import-preview-placeholder-${field.path}`}
                   />
-                </div>
+                )}
               </div>
-            }
-          />
-        ))}
+            </>
+          )}
+        </div>
+      }
+    />
+  );
+}
+
+function ImportPreview({
+  fields,
+  values,
+  onFieldCheckedChanged,
+  setFieldType,
+  loaded,
+  analyzed,
+}: {
+  fields: Field[];
+  values: string[][];
+  onFieldCheckedChanged: (fieldPath: string, checked: boolean) => void;
+  setFieldType: (fieldPath: string, fieldType: string) => void;
+  loaded: boolean;
+  analyzed: boolean;
+}) {
+  if (!loaded) {
```

Comment:
Will this take long ever? Should we show a `SpinLoader`?

Reply:
No longer than before. It just looks at the first row of the CSV file to get the columns and the first 10 rows or so to get the preview. Which was already there before. Maybe?

Reply:
(If yes, then we could just do a follow-up since I'd say this is exactly as before)

Reply:
If it's an easy add and looks alright I think it makes sense, then we're fully transparent to folks as to whats happening. If it's flashing in then out though I think it's fine as is. Your call, feel free to ignore.

Reply:
I think it is pretty much guaranteed to take only a split second. Often with our loaders we end up adding some kind of delay before they appear to stop the flashing anyway. So I'm not sure how often that's going to be an improvement. Not that I have much of an opinion.

Reply:
Filing a follow-up for this.